db.go

```diff
@@ -93,6 +94,10 @@ var defaultNodeKey = route.Vertex{}
 type Db struct {
 	db *sql.DB
 
+	// clearHistory is used to once-off clean up forwarding history in the
+	// case that fwdHistoryLimit is zero. It is included to save making
+	// unnecessary db queried to delete entries that aren't there.
+	clearHistory    sync.Once
```

Comment:
Hmm, maybe it's simpler to just run the cleaner once in `NewDb`, for any value of `fwdHistoryLimit`?